packages/client/src/components/GameEndButton/GameEndButton.tsx

```diff
@@ -1,18 +1,31 @@
+import { memo } from 'react';
 import { Link } from 'react-router-dom';
 import './GameEndButton.css';
 
 interface Props {
   color: string;
   link: string;
   name: string;
+  onClick?: () => void;
 }
 
-export default function GameEndButton(props: Props) {
+function GameEndButton(props: Props) {
+  const { onClick } = props;
+
+  const handleClick = () => {
+    if (onClick) {
+      onClick();
+    }
+  };
```

Comment:
тут же вроде как useCallback нужен?

Reply:
ооо. можно, кстати, красивее

```suggestion
  const handleClick = () => {
    onClick?.()
  };
```